### DISCUSSION TITLE
Github

### DISCUSSION THREAD

**Discussion by BenjaSSJ507** [hackatomik/SISCOx256#32] [Chill | Carrera]
En github se puede programar con C?

**Comment by quantumquirkxyz** [comment]
Aquí tienes una lista de ejemplos de lo que puedes programar en C y alojar en GitHub, creeme puedes hacer mucho más que esto, ya que apenas son algunas cosas 

1. **Sistemas Operativos**: 
   - Puedes programar componentes básicos de un sistema operativo, como un kernel, controladores de hardware, o un sistema de archivos.
   - Ejemplo: Un mini kernel o un sistema operativo simple.

2. **Compiladores o Interpretes**:
   - Programar un compilador para un lenguaje de programación personalizado o un intérprete.
   - Ejemplo: Un compilador simple para un lenguaje de programación inventado.

3. **Algoritmos y Estructuras de Datos**:
   - Implementar algoritmos clásicos como ordenamiento, búsqueda, o estructuras como pilas, colas, árboles y grafos.
   - Ejemplo: Librería de estructuras de datos en C (listas enlazadas, pilas, colas).

4. **Juegos Simples**:
   - Crear juegos simples como Tic-Tac-Toe, Snake, Pong o incluso algo más avanzado como un motor de juego básico.
   - Ejemplo: Un juego de consola como el clásico Snake.

5. **Aplicaciones de Red (Sockets)**:
   - Implementar servidores y clientes para enviar y recibir datos a través de redes (TCP/UDP).
   - Ejemplo: Un servidor HTTP simple o una aplicación de chat en red.

6. **Herramientas de Línea de Comando**:
   - Programar herramientas para la terminal, como un analizador de archivos, calculadoras o herramientas de administración de sistemas.
   - Ejemplo: Una herramienta de gestión de archivos en la terminal o un procesador de texto.

7. **Sistemas de Control (Robótica/IoT)**:
   - Programar controladores para dispositivos IoT o sistemas embebidos como microcontroladores.
   - Ejemplo: Un programa en C para controlar sensores o actuadores de un sistema embebido.

8. **Simulaciones Científicas**:
   - Crear programas que realicen simulaciones numéricas o científicas.
   - Ejemplo: Simulaciones de física o matemáticas (como ecuaciones diferenciales).

9. **Librerías**:
   - Programar una librería para realizar tareas específicas, como manejo de archivos, manipulación de imágenes o cifrado.
   - Ejemplo: Una librería para manejar archivos CSV o para manipular imágenes en formato BMP.

10. **Sistemas de Bases de Datos**:
    - Implementar un motor de base de datos simple.
    - Ejemplo: Un pequeño sistema de base de datos en memoria.

11. **Emuladores**:
    - Programar emuladores para hardware antiguo o sistemas de videojuegos.
    - Ejemplo: Un emulador de una consola de videojuegos clásica como el Game Boy.

12. **Criptografía**:
    - Implementar algoritmos de cifrado y descifrado.
    - Ejemplo: Un programa en C que implemente AES o RSA para cifrado.

13. **Motores de Física**:
    - Desarrollar motores que simulen comportamientos físicos como colisiones o gravedad.
    - Ejemplo: Un motor de física 2D para simulaciones de colisiones entre objetos.

14. **Inteligencia Artificial y Machine Learning**:
    - Implementar algoritmos básicos de AI como búsqueda en árboles, o incluso redes neuronales simples.
    - Ejemplo: Un programa en C que implemente el algoritmo de búsqueda A* o un perceptrón.

15. **Compresión de Datos**:
    - Desarrollar compresores y descompresores para archivos usando algoritmos como Huffman o LZW.
    - Ejemplo: Un programa en C que comprime archivos usando compresión Huffman.

Todos estos ejemplos pueden ser gestionados en GitHub, lo que te permitirá colaborar, mantener versiones de tu código y compartirlo con la comunidad.
